dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -226,3 +230,80 @@ func (cv *CandidateValidation) validateFromChainState(msg ValidateFromChainState
 		Data: *result,
 	}
 }
+
+// precheckPvF prechecks the parachain validation function by retrieving the validation code from the runtime instance
+// and calling the precheck method on the pvf host. It returns the precheck outcome.
+func (cv *CandidateValidation) precheckPvF(relayParent common.Hash, validationCodeHash parachaintypes.
+	ValidationCodeHash) PreCheckOutcome {
+	runtimeInstance, err := cv.BlockState.GetRuntime(relayParent)
+	if err != nil {
+		logger.Errorf("failed to get runtime instance: %w", err)
+		return PreCheckOutcomeFailed
```

Comment:
Why is there both `Failed` and `Invalid`? How do you expect those values to be handled differently in the future?